apps/studio/src/lib/data/tools.ts

```diff
@@ -40,11 +41,17 @@ export const Mutators = {
    * @param  {any} value
    * @returns JsonFriendly
    */
-  genericMutator(value: any, preserveComplex = false): JsonFriendly {
-    const mutate = Mutators.genericMutator
+  genericMutator(dataType: string, dialect: Dialect, value: any, preserveComplex = false): JsonFriendly {
+    const mutate = Mutators.genericMutator.bind(this, dataType, dialect)
     if (_.isBuffer(value)) return value.toString('hex')
     if (_.isDate(value)) return value.toISOString()
-    if (_.isArray(value)) return preserveComplex? value.map((v) => mutate(v, preserveComplex)) : JSON.stringify(value)
+    if (_.isArray(value)) {
+      if (preserveComplex) return value.map((v) => mutate(v, preserveComplex))
```

Comment:
I think this file has changed my mind. I think we should show arrays in the front end as looking like JSON arrays `[a, b, c]`, and convert them in the drivers when writing to the database.

Having them appear as db-specific strings just moves too much database-knowledge into the front-end.

This will also help with data importing where we need to convert arrays for insert.

Reply:
That makes sense.